src/plugins/ui_feature/autofill.ts

```diff
@@ -272,11 +273,19 @@ export class AutofillPlugin extends UIPlugin {
     const activePosition = this.getters.getActivePosition();
 
     const table = this.getters.getTable(activePosition);
-    const row = table ? table.range.zone.bottom : this.getAutofillAutoLastRow();
+    let autofillRow = table ? table.range.zone.bottom : this.getAutofillAutoLastRow();
 
+    // Stop autofill at the next non-empty cell
     const selection = this.getters.getSelectedZone();
-    if (row !== selection.bottom) {
-      this.select(activePosition.col, row);
+    for (let row = selection.bottom + 1; row <= autofillRow; row++) {
+      if (this.getters.getEvaluatedCell({ ...activePosition, row }).type !== CellValueType.empty) {
```

Comment:
I'm afraid you have to check the core cell because a function can return an empty evaluated cell (think about odoo pivot functions) and you don't want the formula to be overwritten in this case.